xsensTFBridge/src/xsensTFBridge.cpp

```diff
@@ -0,0 +1,257 @@
+#include <iostream>
+#include <iomanip>
+#include <limits.h>
+#include <cmath>
+#include <string>
+#include <vector>
+#include <yarp/os/Network.h>
+#include <yarp/os/RFModule.h>
+#include <yarp/os/all.h>
+#include <yarp/os/Wire.h>
+#include <yarp/os/idl/WireTypes.h>
+
+using namespace std;
+using namespace yarp::os;
+
+#include "msgs/String.h"
+#include "std_msgs_Header.h"
+#include "TickTime.h"
+#include "geometry_msgs_Vector3.h"
+#include "geometry_msgs_Quaternion.h"
+#include "geometry_msgs_TransformStamped.h"
+#include "tf2_msgs_TFMessage.h"
+#include <thrift/XsensSegmentsFrame.h>
+#include "thrift/XsensDriverService.h"
+
+inline TickTime normalizeSecNSec(double yarpTimeStamp)
+{
+    uint64_t time = (uint64_t) (yarpTimeStamp * 1000000000UL);
+    uint64_t nsec_part = (time % 1000000000UL);
+    uint64_t sec_part = (time / 1000000000UL);
+    TickTime ret;
+
+    if (sec_part > UINT_MAX)
+    {
+        yWarning() << "Timestamp exceeded the 64 bit representation, resetting it to 0";
+        sec_part = 0;
+    }
+
+    ret.sec  = sec_part;
+    ret.nsec = nsec_part;
+    return ret;
+}
+
+class xsensTfPublisherModule : public RFModule, public yarp::os::TypedReaderCallback<xsens::XsensSegmentsFrame> 
+{
+    Port client_port;
+    yarp::os::BufferedPort<xsens::XsensSegmentsFrame> xsensDataPort;
+    xsens::XsensDriverService xsensDriver;
+    
+    Publisher<tf2_msgs_TFMessage> publisher_tf;
+    
+    //variables declaration
+    int i, segmentsCount, trueSegmentsCount, offline;
+    tf2_msgs_TFMessage tf;
+    std::vector<std::string> segments;
+    std::vector<xsens::FrameReferece> xsensSegments;
+    
+public:
+    double getPeriod()
+    {
+        // module periodicity (seconds).
+        return 100;
+    }
+    // Main function. Will be called periodically every getPeriod() seconds
+    bool updateModule()
+    {
+        return true;
+    }
+    // Message handler. 
+    bool respond(const Bottle& command, Bottle& reply)
+    {
+        if (command.get(0).asString() == "quit")
+        return false;
+        else
+        reply = command;
+        return true;
+    }
+    // Configure function. 
+    bool configure(yarp::os::ResourceFinder &rf)
+    {
+        std::string moduleName = rf.check("name", yarp::os::Value("xsensTFBridge"), "Checking module name").asString();
+        std::string serverName = rf.check("serverName", yarp::os::Value("xsens"), "Checking module name").asString();
+        setName(moduleName.c_str());
+        
+        if (!publisher_tf.topic("/tf")) {
+            cerr<< "Failed to create publisher to /tf\n";
+            return false;
+        }
+        
+        i = 0;
```

Comment:
Differently from old versions of C, it is now considered a good code style to declare variables next to their use.
Furthermore I see two issues on this variable `i`:
- either it is a local variable (needed only in this part of code)
- or you cannot give as a name to a member variable `i`. It is too ambiguous!